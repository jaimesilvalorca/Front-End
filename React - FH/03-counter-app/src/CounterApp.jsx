import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value)

const handleAdd = ()=>{
    setCounter(counter+1)
}

const handleRest = ()=>{
    setCounter(counter-1)
}

const handleReset = ()=>{
    setCounter(value)
}


  return (
    <>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>

        <button onClick={handleAdd}>
            +1
        </button>
        <button onClick={handleRest}>
            -1
        </button>
        <button aria-label='btn-reset' onClick={handleReset}>
            reset
        </button>
    </>
  )
}

CounterApp.propTypes ={
    value:PropTypes.number.isRequired
}