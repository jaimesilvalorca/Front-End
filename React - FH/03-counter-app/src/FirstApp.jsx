import React from 'react'
import PropTypes from 'prop-types';


const newMessage = {
    message:'Hola mundo',
    title:'Jaime'
};

function saludo(saludo){
    return saludo
}

const newArray = [1,2,3,4,5,6,7,8,9]

export const FirstApp = ({title,subTitle,name}) => {

    
    return (
        <>
            <div data-testid="test-title">{title}</div>
            <h1>{subTitle}</h1>
            <h1>{name}</h1>
        </>

    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle:PropTypes.string,
    name:PropTypes.string


}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subTitle:'No hay subtitutlo',
    // name:'Jaime Silva'

}