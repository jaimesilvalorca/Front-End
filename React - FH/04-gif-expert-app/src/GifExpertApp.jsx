import { useState } from "react"
import { AddCategory,GifGrid  } from "./components"

export const GifExpertApp = () => {


  const [categories, setCategories] = useState(['One Punch'])


  const onAddCategory = (onNewCategory) => {
    console.log(categories)
    if (categories.includes(onNewCategory)) return;

    setCategories([onNewCategory,...categories, ])
    // setCategories(cat=>[...cat,onNewCategory])



  }

  return (
    <>
      {/* titulo */}
      <h1>Gif expert app</h1>
      {/* Input */}
      <AddCategory
        onNewCategory={event => onAddCategory(event)}
      />

      {/* Listado de Gif */}

      {
        categories.map((category) => (
          <GifGrid
            key={category}
            category={category}
          />



        ))}
    </>
  )
}
