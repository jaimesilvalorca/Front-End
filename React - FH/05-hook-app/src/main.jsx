import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
// import { HooksApp } from './HooksApp'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterWithCustomHooks } from './01-useState/CounterWithCustomHooks'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { Layout } from './05-useLayoutEffect/Layout'
// import { Memorize } from './06-memos/Memorize'
// import { MemoHook } from './06-memos/MemoHook'
// import { CallbackHook } from './06-memos/CallbackHook'
// import TodoApp from './08-useReducer/TodoApp'
// import { Padre } from './07-tarea-memo/Padre'
// import './08-useReducer/intro-reducer'
import { MainApp } from './09-useContext/MainApp'
import { BrowserRouter } from 'react-router-dom'

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    <React.StrictMode>
      <MainApp />
    </React.StrictMode>
  </BrowserRouter>
)
