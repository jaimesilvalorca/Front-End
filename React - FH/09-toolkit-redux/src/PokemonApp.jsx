import { useEffect } from "react"
import {useDispatch,useSelector} from 'react-redux'
import { getPokemons } from "./store/slices/pokemon";


const PokemonApp = () => {

  const dispatch = useDispatch();

  
  const {isLoading,pokemons,page} = useSelector(state=> state.pokemons)
  // const {isLoading} = useSelector(state=> state.isLoading)
  useEffect(() => {
    dispatch(getPokemons());  

  }, [dispatch])
  

  
  
  return (
    <>
    <h1>Pokemon App</h1>
    <hr />
    <span>Loading:{ isLoading ? 'true': 'false'}</span>
    <ul>
    {
      pokemons.map(pokemon=>
        <li key={pokemon.url}>{pokemon.name}</li>)
    }
     </ul>
   
    <button
    onClick={()=>dispatch(getPokemons(page))}
    >
Siguiente
    </button>
    

    </>

  )
}

export default PokemonApp