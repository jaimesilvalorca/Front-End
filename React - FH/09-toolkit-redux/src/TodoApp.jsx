import { useState } from "react"
import { useGetTodoQuery } from "./store/apis"


export const TodoApp = () => {

    // const {data:todos = [],isLoading} = useGetTodosQuery()

    const [todoId, setTodoId] = useState(1)
    const {data:todo,isLoading} = useGetTodoQuery(todoId)
    console.log(todo)

    const nextTodo = () =>{
        setTodoId(todoId +1 )
    }

    const prevTodo= ()=>{
        if(todoId === 1) return
        setTodoId(todoId-1)
    }


  return (
    <>
    <h1>Todos - RTK QUery</h1>
    <hr />
    <h4>isLoading: {isLoading? 'true' :'false'}</h4>
    <pre>{JSON.stringify(todo)}</pre>
    {/* <ul>
        {todo.map(todo=>(
            
            <li key={todo.id}>
                <strong>{todo.completed ? 'DONE':'PENDING'} </strong>    
                {todo.title}
                </li>
        ))}
    </ul> */}
    <button
    onClick={()=>nextTodo()}
    >
        next Todo
    </button>
    <button
    onClick={()=>prevTodo()}
    >
        prev Todo
    </button>
    </>
  )
}
