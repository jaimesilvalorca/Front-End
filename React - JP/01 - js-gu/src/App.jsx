import { useState,useEffect} from "react"
import { Guitar } from "./components/Guitar"
import { Header } from "./components/Header"
import { db } from "./data/db"

function App() {

  const initialCart = ()=>{
    const localStorageCart = localStorage.getItem('cart')

    return localStorageCart ? JSON.parse(localStorageCart) : []
  }
  
  const [data] = useState(db)
  const [cart, setCart] = useState(initialCart)

  const MAX_ITEMS = 5
  const MIN_ITEMS = 1

  useEffect(()=>{
      localStorage.setItem('cart',JSON.stringify(cart))

  
  },[cart])

  function addToCart(item){
    console.log('agregando....')
    const itemExists = cart.findIndex((guitar)=> guitar.id === item.id)

    if(itemExists >=0){
      if(cart[itemExists].quantity>=MAX_ITEMS) return
      const updatedCart = [...cart]
      updatedCart[itemExists].quantity++
      setCart(updatedCart)
    }else{
      console.log('no existe agregando')
      item.quantity = 1
      setCart([...cart,item])
    }

    saveLocalStorage()
    

  }

  function removeFromCart(id){
    setCart(prevCart=>prevCart.filter(guitar => guitar.id !== id))
  }

  function increaseQuantity(id){
    const updatedCart = cart.map(item=>{
      if(item.id === id && item.quantity < MAX_ITEMS){
        return{
          ...item,
          quantity:item.quantity+1
        }
      }
      return item
    })
    setCart(updatedCart)
  }

  function decreaseQuantity(id){
    const updatedCart = cart.map(item=>{
      if(item.id === id && item.quantity > MIN_ITEMS){
        return{
          ...item,
          quantity:item.quantity-1
        }
      }
      return item
    })
    setCart(updatedCart)
  }

  function clearCart(){
    setCart([])
  }

  function saveLocalStorage(){
    localStorage.setItem('cart',JSON.stringify(cart))
  }

  return (
    <>
      <Header 
      cart={cart}
      clearCart={clearCart}
      removeFromCart={removeFromCart}
      increaseQuantity={increaseQuantity}
      decreaseQuantity={decreaseQuantity}
      
      />
      <main className="container-xl mt-5">
        <h2 className="text-center">Nuestra Colecci√≥n</h2>

        <div className="row mt-5">
          {
            data.map((guitar)=>(
                  <Guitar
                  cart={cart}
                  key={guitar.id}
                  guitar={guitar}
                  setCart={setCart}
                  addToCart={addToCart}
                  
                  />
           
              

            ))}

          
        </div>
      </main>


      <footer className="bg-dark mt-5 py-5">
        <div className="container-xl">
          <p className="text-white text-center fs-4 mt-4 m-md-0">GuitarLA - Todos los derechos Reservados</p>
        </div>
      </footer>
    </>
  )
}

export default App
