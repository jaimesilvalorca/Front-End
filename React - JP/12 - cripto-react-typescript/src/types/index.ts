import {CryptoCurrencyResponseSchema, CurrencySchema} from '../schema/crypto-schema'
import {z} from 'zod'

export type Currency = z.infer<typeof CurrencySchema>

export type Cryptocurrency = z.infer<typeof CryptoCurrencyResponseSchema>