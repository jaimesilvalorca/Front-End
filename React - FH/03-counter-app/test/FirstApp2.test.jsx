import { getAllByText, getByText, render,screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp',()=>{

    const title = 'hola soy goku'
    const subTitle = 'Soy un subtitutlo'

    test('Debe de hacer match con el snapshot',()=>{

        

        const {container} = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar el mensaje "Hola soy Goku',()=>{

        render(<FirstApp title={title}/>)
        expect(screen.getByText(title)).toBeTruthy()
    })

    // test('debe mostrar el titutlo en un h1',()=>{
    //     render(<FirstApp title={title}/>)
    //     expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title)
    // })


    test('debe mostrar el subtitulo enviado por props',()=>{
        
        render(
        <FirstApp 
        title={title} 
        subTitle={subTitle}
        />
        )

        expect(screen.getAllByText(subTitle).length).toBe(1)
    })

})