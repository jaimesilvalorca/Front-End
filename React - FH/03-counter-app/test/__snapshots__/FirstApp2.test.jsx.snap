// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    hola soy goku
  </div>
  <h1>
    No hay subtitutlo
  </h1>
  <h1 />
</div>
`;
